docs/clustering.md

```diff
@@ -1,92 +1,168 @@
-  * Organizations
-  * Inventories
-  * Job Templates
+```
+       ┌───────────────────────────┐
+       │      Load-balancer        │
+       │   (configured separately) │
+       └───┬───────────────────┬───┘
+           │   round robin API │
+           ▼       requests    ▼
+
+  AWX VM/Pod 1               AWX VM/Pod 2
+┌──────────────┐           ┌──────────────┐
+│              │           │              │
+│ ┌──────────┐ │           │ ┌──────────┐ │
+│ │ awx-task │ │           │ │ awx-task │ │
+│ ├──────────┤ │           │ ├──────────┤ │
+│ │ awx-ee   │ │           │ │ awx-ee   │ │
+│ ├──────────┤ │           │ ├──────────┤ │
+│ │ awx-web  │ │           │ │ awx-web  │ │
+│ ├──────────┤ │           │ ├──────────┤ │
+│ │ redis    │ │           │ │ redis    │ │
+│ └──────────┘ │           │ └──────────┘ │
+│              │           │              │
+└──────────────┴─────┬─────┴──────────────┘
+                     │
+                     │
+               ┌─────▼─────┐
+               │ Postgres  │
+               │ database  │
+               └───────────┘
+```
+
+There are two main deployment types, virtual machines (VM) or K8S. Ansible Automation Controller installation supports both VM or K8S deployments. The upstream AWX project only supports K8S deployments. Either deployment type supports cluster scaling.
+- Control plane nodes run a number of background services that are managed by supervisord
+  - dispatcher
+  - wsbroadcast
+  - callback receiver
+  - receptor (*managed under systemd)
+  - redis (*managed under systemd)
+  - uwsgi
+  - daphne
+  - rsyslog
+- For K8S deployments, these background processes are containerized
+  - `awx-ee`: receptor
+  - `awx-web`: uwsgi, daphne, wsbroadcast, rsyslog
+  - `awx-task`: dispatcher, callback receiver
+  - `redis`: redis
+- Each control node is monolithic and contains all the necessary components for handling API requests and running jobs.
+- A load balancer in front of the cluster can handle incoming web requests and send them control nodes based on load balancing rules (e.g. round robin)
+- AWX supports a single, shared Postgres database
+- AWX is configured in such a way that if any of these services or their components fail, then all services are restarted. If these fail sufficiently (often in a short span of time), then the entire instance will be placed offline in an automated fashion in order to allow remediation without causing unexpected behavior.
+
+## Scaling the cluster
+
+For VM (AAP) deployments, scaling up involves modifying `inventory`and re-running setup.sh
+For K8s deployments, scaling up is handled by changing the number of replicas in the AWX replica set.
+
+After scaling up, the new control plane node is registered in the database as a new `Instance`.
 
-  ...such that execution of jobs under those resources will favor particular queues.
+Instance types:
+`hybrid` - control plane node that can also run jobs
```

Comment:
Please mention that hybrid and hop nodes are not available in default installations of AWX.

Reply:
I made it clear these are part of AAP only